From 40645afa36c94f52920815f57cec8e44befecab8 Mon Sep 17 00:00:00 2001
From: saravanj24 <saravanan.j@jasmin-infotech.com>
Date: Thu, 24 Aug 2023 14:20:16 +0530
Subject: [PATCH] 0005-Setting-U-Boot-env-to-load-rootfs-from-initramfs-6sx

---
 include/configs/mx6sxsabresd.h | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/include/configs/mx6sxsabresd.h b/include/configs/mx6sxsabresd.h
index a0cabe565e..60c04159b6 100644
--- a/include/configs/mx6sxsabresd.h
+++ b/include/configs/mx6sxsabresd.h
@@ -84,6 +84,9 @@
 	"fdt_addr=0x83000000\0" \
 	"tee_addr=0x84000000\0" \
 	"tee_file=uTee-6sxsdb\0" \
+	"live_initramfs_file=core-image-minimal-live-initramfs-imx6sxsabresd.cpio.gz\0" \
+	"mfg_initramfs_file=core-image-minimal-mfg-initramfs-imx6sxsabresd.cpio.gz\0" \
+	"prod_initramfs_file=core-image-minimal-prod-initramfs-imx6sxsabresd.cpio.gz\0" \
 	"boot_fdt=try\0" \
 	"ip_dyn=yes\0" \
 	"splashimage=0x8c000000\0" \
@@ -100,6 +103,30 @@
 	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
 	"loadtee=fatload mmc ${mmcdev}:${mmcpart} ${tee_addr} ${tee_file}\0" \
+	"loadliveinitramfs=fatload mmc ${mmcdev}:${mmcpart} ${initrd_addr} ${live_initramfs_file}\0" \
+	"loadmfginitramfs=fatload mmc ${mmcdev}:${mmcpart} ${initrd_addr} ${mfg_initramfs_file}\0" \
+	"loadprodinitramfs=fatload mmc ${mmcdev}:${mmcpart} ${initrd_addr} ${prod_initramfs_file}\0" \
+	"liveinitramfskernelboot=echo Booting live initramfs  . . .;" \
+                           "setenv fdt_file imx6sx-sdb.dtb; " \
+			    "run mmcargs; " \
+			    "run loadimage; " \
+			    "run loadfdt; " \
+			    "run loadliveinitramfs; " \
+			    "bootz ${loadaddr} ${initrd_addr}:${filesize} ${fdt_addr};\0" \
+         "mfginitramfskernelboot=echo Booting mfg initramfs  . . .;" \
+                            "setenv fdt_file imx6sx-sdb.dtb; " \
+                            "run mmcargs; " \
+                            "run loadimage; " \
+                            "run loadfdt; " \
+                            "run loadmfginitramfs; " \
+                            "bootz ${loadaddr} ${initrd_addr}:${filesize} ${fdt_addr};\0" \
+	"prodinitramfskernelboot=echo Booting production initramfs  . . .;" \
+                            "setenv fdt_file imx6sx-sdb.dtb; " \
+                            "run mmcargs; " \
+                            "run loadimage; " \
+                            "run loadfdt; " \
+                            "run loadprodinitramfs; " \
+                            "bootz ${loadaddr} ${initrd_addr}:${filesize} ${fdt_addr};\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${tee} = yes; then " \
